src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,244 @@
+            }
+
+            new Deadline(splitTime[0], splitTime[1], false);
+            break;
+        case "event":
+            if (splitTask.length == 1) {
+                emptyDescription = true;
+                break;
+            }
+
+            String[] splitTimeEvent = splitTask[1].split("/", 2);
+            if (splitTimeEvent.length == 1) {
+                emptyTimelineError = "event";
+                break;
+            }
+
+            new Event(splitTimeEvent[0], splitTimeEvent[1], false);
+            break;
+        default:
+            throw new DukeException.InvalidCommandException(input);
+        }
+
+        if (emptyDescription) {
+            throw new DukeException.EmptyDescriptionException();
+        } else if (emptyTimelineError != null) {
+            throw new DukeException.EmptyTimelineDescription(emptyTimelineError);
+        }
+    }
+
+    /**
+     * This method is responsible for creating the task based on the file storage contents. This
+     * method differs from the original add method as it does not go through any checks and parsing
+     * is already done.
+     *
+     * @param taskChar the identifier for the type of task class to be created
+     * @param taskStatus the identifier for the status of task class to be created
+     * @param description the description of the task
+     * @param timeline the timeline of the task
+     * @return the new task created based on the given parameters
+     */
+    public TaskList addExisting(char taskChar, char taskStatus, String description, String timeline) {
+        TaskList currentTask;
+
+        if (taskChar == 'T') {
+            currentTask = new Todo(description, true);
+        } else if (taskChar == 'D') {
+            currentTask = new Deadline(description, timeline, true);
+        } else {
+            currentTask = new Event(description, timeline, true);
+        }
+
+        if (taskStatus == 'X') {
+            currentTask.setDone();
+        }
+        return currentTask;
+    }
+
+    /**
+     * This method is in charge of loading the contents of the file storage into the temporary
+     * arraylist that holds the tasks in order. This ensures that the user is always interacting
+     * with the contents of the storage file at the start.
+     */
+    public void loadArrayList() {
+        FILE.loadFile();
+        FILE.overwriteList(TASKS);
+    }
+
+    /**
+     * This method retrieves the task based on the index in the task list.
+     *
+     * @param index the index of the task in the task list to be retrieved
+     * @return the specified task retrieved from the task list
+     */
+    public static TaskList getTask(int index) {
+        if (index < 0 || index >= TASKS.size()) {
+            USER_INTERFACE.invalidIntegerWarning();
+            return null;
+        } else {
+            return TASKS.get(index);
+        }
+    }
+
+    /**
+     * This method finds the tasks that matches the input keyword and subsequently
```

Comment:
Not sure if this is required/necessary. Perhaps you can consider to start the Javadoc comments with `Finds the tasks ...` instead of `This method...` ? I have also noticed it in some other Javadoc comments as well.
Also, I have noticed that 2 styles are used for Javadoc comments, namely starting with "This method" and not starting with "This method". Perhaps you can consider to standardize it ?

Reply:
I agree with you. It should be phrased in a different manner. Thanks for pointing that out and I will make the necessary changes.